nat-traversal/protocol.go

```diff
@@ -36,47 +36,47 @@ const (
-	reportNATTraversalResultDetailHandler func(err error, packet nex.PacketInterface, callID uint32, cid uint32, result bool, detail int32, rtt uint32) uint32
+	Server                         nex.ServerInterface
+	RequestProbeInitiation         func(err error, packet nex.PacketInterface, callID uint32, urlTargetList []*nex.StationURL) (*nex.RMCMessage, uint32)
+	InitiateProbe                  func(err error, packet nex.PacketInterface, callID uint32, urlStationToProbe *nex.StationURL) (*nex.RMCMessage, uint32)
+	RequestProbeInitiationExt      func(err error, packet nex.PacketInterface, callID uint32, targetList []string, stationToProbe string) (*nex.RMCMessage, uint32)
+	ReportNATTraversalResult       func(err error, packet nex.PacketInterface, callID uint32, cid uint32, result bool, rtt uint32) (*nex.RMCMessage, uint32)
+	ReportNATProperties            func(err error, packet nex.PacketInterface, callID uint32, natmapping uint32, natfiltering uint32, rtt uint32) (*nex.RMCMessage, uint32)
+	GetRelaySignatureKey           func(err error, packet nex.PacketInterface, callID uint32) (*nex.RMCMessage, uint32)
+	ReportNATTraversalResultDetail func(err error, packet nex.PacketInterface, callID uint32, cid uint32, result bool, detail int32, rtt uint32) (*nex.RMCMessage, uint32)
 }
 
 // Setup initializes the protocol
 func (protocol *Protocol) Setup() {
-	protocol.Server.On("Data", func(packet nex.PacketInterface) {
-		request := packet.RMCRequest()
+	protocol.Server.OnData(func(packet nex.PacketInterface) {
+		request := packet.RMCMessage()
 
-		if request.ProtocolID() == ProtocolID {
-			switch request.MethodID() {
+		if request.ProtocolID == ProtocolID {
+			switch request.MethodID {
 			case MethodRequestProbeInitiation:
-				go protocol.handleRequestProbeInitiation(packet)
+				protocol.handleRequestProbeInitiation(packet)
 			case MethodInitiateProbe:
-				go protocol.handleInitiateProbe(packet)
+				protocol.handleInitiateProbe(packet)
 			case MethodRequestProbeInitiationExt:
-				go protocol.handleRequestProbeInitiationExt(packet)
+				protocol.handleRequestProbeInitiationExt(packet)
 			case MethodReportNATTraversalResult:
-				go protocol.handleReportNATTraversalResult(packet)
+				protocol.handleReportNATTraversalResult(packet)
 			case MethodReportNATProperties:
-				go protocol.handleReportNATProperties(packet)
+				protocol.handleReportNATProperties(packet)
 			case MethodGetRelaySignatureKey:
-				go protocol.handleGetRelaySignatureKey(packet)
+				protocol.handleGetRelaySignatureKey(packet)
 			case MethodReportNATTraversalResultDetail:
-				go protocol.handleReportNATTraversalResultDetail(packet)
+				protocol.handleReportNATTraversalResultDetail(packet)
 			default:
 				go globals.RespondError(packet, ProtocolID, nex.Errors.Core.NotImplemented)
-				fmt.Printf("Unsupported NATTraversal method ID: %#v\n", request.MethodID())
+				fmt.Printf("Unsupported NATTraversal method ID: %#v\n", request.MethodID)
```

Comment:
We aren't checking if the RMC message is a request, which makes the MK7 server show logs about `InitiateProbe` not being implemented (since the server isn't expected to receive a request on that method). This doesn't affect this game, but it may affect others since we are sending unexpected "not implemented" replies

Reply:
How was this working before?? O.o

I don't think we checked if it was a request before either?? o.O